effekt/jvm/src/test/scala/effekt/PolymorphismBoxingTests.scala

```diff
@@ -0,0 +1,286 @@
+package effekt
+
+import java.io.File
+import effekt.core.{Block, Definition, DirectApp, PolymorphismBoxing, Pure, Run, Stmt}
+import effekt.context.Context
+import effekt.source.{IdDef, Import, ModuleDecl}
+import kiama.{parsing, util}
+import effekt.symbols.{Module, Name, TypeConstructor, TypeSymbol, ValueSymbol, ValueType}
+import effekt.source
+import effekt.util.messages
+import effekt.util.messages.DebugMessaging
+import kiama.parsing.{Failure, NoSuccess, Success}
+import kiama.util.Severities
+
+abstract class AbstractPolymorphismBoxingTests extends munit.FunSuite {
+
+  def boxDef(tpe: ValueType.ValueTypeApp): List[symbols.Symbol] = {
+    val tpeCns: symbols.TypeConstructor.Record =
+      symbols.TypeConstructor.Record(symbols.LocalName("Boxed" ++ tpe.constructor.name.name), List(), null)
```

Comment:
`null`??? really?

Reply:
I know why you do this, but ideally we wouldn't need to cook up some symbols, but only add declarations to the core module.

Reply:
Maybe you can add the symbols to `builtins` for now? WDYT?

Reply:
We could do that, yes (We would still temporarily use something like null to be able to tie the knot, though).

The main reason, why I used symbols here is, that the `Boxed*` types should be part of the stdlib module, and this way, we can run this phase separately on the modules (before `Aggregate`).

Reply:
Currently, they are defined here: https://github.com/effekt-lang/effekt/pull/199/files/f50fb68cac88083e264fd21b96cfbaf0a61c89d4#diff-feddea404f7474369f1a93d26ee304085c61ce17003f0ea38c076fc30cd6e118R162-R166
and will then be searched in the prelude module (by name); this is generating them for testing.

Reply:
That raises a general question: Not relying on symbols, where do we get the declarations for imported types/... when working on core? Or should the declarations contain _all_ declarations and not only the ones from this module? (In that case: How can we differentiate between imported and other declarations?)